lib/corefoundation/base_wrapper.rb

```diff
@@ -0,0 +1,112 @@
+# The top level namespace for the corefoundation library. The raw FFI generated methods are attached here
+module CF
+  extend FFI::Library
+  ffi_lib "/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation"
+
+  if FFI::Platform::ARCH == "x86_64"
+    typedef :long_long, :cfindex
+    typedef :long_long, :cfcomparisonresult
+    typedef :ulong_long, :cfoptionflags
+    typedef :ulong_long, :cftypeid
+    typedef :ulong_long, :cfhashcode
+  else
+    typedef :long, :cfindex
+    typedef :long, :cfcomparisonresult
+    typedef :ulong, :cfoptionflags
+    typedef :ulong, :cftypeid
+    typedef :ulong, :cfhashcode
+  end
+
+  typedef :pointer, :cftyperef
+
+  attach_function :show, "CFShow", [:cftyperef], :void
+  attach_function :release, "CFRelease", [:cftyperef], :void
+  attach_function :retain, "CFRetain", [:cftyperef], :cftyperef
+  attach_function "CFEqual", %i{cftyperef cftyperef}, :char
+  attach_function "CFHash", [:cftyperef], :cfhashcode
+  attach_function "CFCopyDescription", [:cftyperef], :cftyperef
+  attach_function "CFGetTypeID", [:cftyperef], :cftypeid
+
+  # @private
+  class Range < FFI::Struct
+    layout :location, :cfindex,
+           :length, :cfindex
+  end
+
+  # Provides a more declarative way to define all required abstract methods.
+  # This gets included in the CF::Base class whose subclasses then need to define those methods.
+  module BaseWrapper
```

Comment:
A descriptive name seems desirable

Reply:
Naming is hard lol. But on a serious note, I haven't changed it yet because this is still a proposal at best. Once we have thoroughly reviewed this new design, I'm hoping I'll have a better name for this.